src/charm.py

```diff
@@ -152,7 +163,64 @@ def temp_maintenance_status(self, message: str):
         yield
         self.unit.status = previous_status
 
-    def _on_collect_status(self, event: CollectStatusEvent):
+    def _on_authorize_charm_action(self, event: ActionEvent):
+        """Authorize the charm to interact with Vault."""
+        if not self.unit.is_leader():
+            event.fail("This action can only be run by the leader unit")
+            return
+        logger.info("Authorizing the charm to interact with Vault")
+        if not self._api_address:
+            event.fail("API address is not available.")
+            return
+        if not self.tls.tls_file_available_in_charm(File.CA):
+            event.fail("CA certificate is not available in the charm. Something is wrong.")
+            return
+        token = event.params["token"]
+        vault = self._get_vault_client()
```

Comment:
I personally despise returning an error string. It feels very unpythonic to me, whatever that means. It's just not a very common pattern in Python AFAIK.

I'm fine with changing it to `None`, but I was actually trying to avoid that as well [as mentioned above](https://github.com/canonical/vault-operator/pull/62#discussion_r1530274205).

But, you made me realize why I dislike all these ways (we're not following very pythonic patterns) and I think have a better approach I'll propose independently. For now I'll just follow the `None` approach as we already do that elsewhere.

Reply:
I've followed the `None` pattern we've been using in https://github.com/canonical/vault-operator/pull/62/commits/4c44771a582c936e8d0d9d9725484dd2121d9a2d -- let me know if that's good enough for now.

Like I mentioned, I'd like to re-do this properly but that's a bigger change to our overall pattern that should probably be kept separate. I don't think returning strings is the right direction forward. We should be using exceptions properly here.

Reply:
I think the pythonic way is to have bigass try catch blocks everywhere, and I hate that more. I like https://github.com/canonical/vault-operator/pull/62/commits/4c44771a582c936e8d0d9d9725484dd2121d9a2d way more.

Reply:
I don't disagree, they're just all gross. I equally dislike the "return object or an error string" -- that feels very Javascript-esq, and I'd hesitate to model anything after Javascript in the name of readability :laughing: 

There are ways to avoid the big try..catch blocks everywhere, it just needs some re-working. I'll try to propose something in a follow-up PR if I can simplify things.

Reply:
It's also very golang-like :D

Reply:
Right, and I've noticed quite a few golang-esq patterns leaking their way in recently.

I love golang, but I think we need to be careful with that.